EIPS/eip-00.md

```diff
@@ -0,0 +1,790 @@
+
+#### `eth_getTransactionReceipt`
+
+Accepts the hash of a virtual transaction that encapsulates the validation frames of the `AA_TX_TYPE` transaction.
+This transaction's ID is defined as (`AA_TX_TYPE transaction hash + 1`).
+
+If an AA transaction is included in a block, returns the following values in addition to the existing fields:
+
+| Name                       | Value                                                                        |
+|----------------------------|------------------------------------------------------------------------------|
+| sender                     | Address of the sender of this transaction                                    |
+| nonce                      | The transaction nonce value                                                  |
+| paymaster                  | Address of the Paymaster if it is paying for the transaction, null otherwise |
+| deployer                   | Address of the Deployer if it is included in the transaction, null otherwise    |
+| senderCreationGasUsed      | The amount of gas actually used by the sender deployment frame               |
+| senderValidationGasUsed    | The amount of gas actually used by the sender validation frame               |
+| paymasterValidationGasUsed | The amount of gas actually used by the paymaster validation frame            |
+
+Accepts hash of Transaction Type `AA_TX_TYPE`.
+
+If an AA transaction is included in a block, returns the following values in addition to the existing fields:
+
+| Name                            | Value                                                                                          |
+|---------------------------------|------------------------------------------------------------------------------------------------|
+| status                          | Either 1 (success) or 0 (failure) status of the execution frame                                |
+| executionGasUsed                | The amount of gas actually used by the execution frame                                         |
+| postPaymasterTransactionStatus  | Either 1 (success), 0 (failure), or `null` (did not run) status of the `postPaymasterTransaction` frame |
+| postPaymasterTransactionGasUsed | The amount of gas actually used by the paymaster `postPaymasterTransaction` frame                       |
+
+Note that the field `to` is not included as there is no clear `target` in an `AA_TX_TYPE` transaction.
+
+#### `eth_call`
+
+Accepts Transaction Type `AA_TX_TYPE` with all fields except `from` and `callData` optional.
+
+Returns the return value of [the `sender` execution frame](#sender-execution-frame).
+
+If provided with `paymasterData` and `deployerData` also executes the corresponding frame.
+
+If any of the frames reverts the call returns the revert data of each reverted frame.
+
+#### `eth_estimateGasAccountAbstraction`
+
+Accepts Transaction Type `AA_TX_TYPE` with fields `validationGasLimit`, `paymasterGasLimit`, `callGasLimit` optional.
```

Comment:
Designing smart contracts that have complicated validation functions but have an easily calculable "dummy signature" has proven to be difficult for many account devs. I've seen instances of this not being possible, and devs just determining a max value for `validationGasLimit` that is consistent with their account types. This makes sense as validation gas consumption shouldn't change much.

I think we should add a switch to this method that lets the node skip these estimations if the user doesn't want them to run.

Possibly if the user supplies any of these 3 limit fields the RPC will skip estimation for that portion and just return the provided value?

Reply:
We can't skip running validation, as it may change the state in ways that affect the transaction.  Besides, validation may check more things besides the signature, so it may revert on chain on one of these checks after being skipped in gas estimation.

The node has to run it for completeness, but the user is free to override the estimated value and use something different.  The RPC could get a number from the user and return that number instead of the actual result, but it seems like just syntactic sugar.  If the user provides the numbers, the user can also use them instead of the returned ones.

Reply:
Looking to save nodes some processing cycles if the user knows how much gas to use for validation. Depending on the account implementation it may be difficult for the user to supply a valid dummy signature that doesn't revert. I think in those cases the node should still be able to return a `callGasLimit`

Reply:
The problem is that the node has no way to know whether some meaningful state change (or more likely a transient state change) happens during validation, which would alter the execution flow of the call itself.  The user may also not know it.

Consider an ERC-6900 account for example:

A 6900 plugin may have both a validation hook and a post-execution hook.  Let's say it's a plugin that supports two signature types: 1. Mobile signature. 2. Ledger signature.  And it enforces different spending limits on the account in post-execution, depending on whether the transaction was signed by the mobile phone or by the ledger. The behavior of the plugin is undefined if it's post-execution was called without going through its validation.

Similarly, if an account that supports session keys associated with different operations, it has no way to determine how a call should behave if it doesn't know which session key signed the transaction.

Reply:
Understood. However, this is a suggestion for simple wallets that have very static validation methods and only for gas estimation. For example it doesn't really make sense to estimate the validation gas for SimpleAccount each time. Users could provide a value to the gas estimation RPC and save some compute cycles (and potentially RPC costs)

Reply:
wallets can cache validation estimation but need to take care, e.g., to invalidate the cache if plugins are modified.
We can only suggest on the non-cached mode.